src/sentry/middleware/ratelimit.py

```diff
@@ -23,20 +23,29 @@
 )
 
 
-class RatelimitMiddleware(MiddlewareMixin):
-    """Middleware that applies a rate limit to every endpoint."""
+class RatelimitMiddleware:
+    """Middleware that applies a rate limit to every endpoint.
+    See: https://docs.djangoproject.com/en/4.0/topics/http/middleware/#writing-your-own-middleware
+    """
+
+    def __init__(self, get_response):
+        self.get_response = get_response
+
+    def __call__(self, request: Request) -> Response:
+        # First, check if the endpoint call will violate.
+        data = {}
 
-    def process_view(self, request: Request, view_func, view_args, view_kwargs) -> Response | None:
-        """Check if the endpoint call will violate."""
         try:
-            # TODO: put these fields into their own object
-            request.will_be_rate_limited = False
             request.rate_limit_category = None
-            request.rate_limit_uid = uuid.uuid4().hex
-            request.rate_limit_key = get_rate_limit_key(view_func, request)
-            if request.rate_limit_key is None:
+            rate_limit_uid = uuid.uuid4().hex
+            view_func = resolve(request.path).func
```

Comment:
since this is slightly unusual, it may be worth leaving a comment as to why this step is being done